avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -959,41 +960,78 @@ private String getMethodNameForFieldWithPrefix(String prefix, String fieldName)
   }
 
   private void addCustomDecodeMethod(MethodSpec.Builder customDecodeBuilder, AvroRecordSchema recordSchema,
-      SpecificRecordGenerationConfig config) {
+      SpecificRecordGenerationConfig config, TypeSpec.Builder classBuilder) {
+    int blockSize = 25, fieldCounter = 0, chunkCounter = 0;
     // reset var counter
     sizeValCounter = -1;
     customDecodeBuilder.addStatement(
             "org.apache.avro.Schema.Field[] fieldOrder = (com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper.areFieldsReordered(getSchema())) ? in.readFieldOrder() : null")
         .beginControlFlow("if (fieldOrder == null)");
-    for(AvroSchemaField field : recordSchema.getFields()) {
-      String escapedFieldName = getFieldNameWithSuffix(field);
-      customDecodeBuilder.addStatement(getSerializedCustomDecodeBlock(config, field.getSchemaOrRef().getSchema(),
-          field.getSchemaOrRef().getSchema().type(), "this." + replaceSingleDollarSignWithDouble(escapedFieldName),
-          "this." + replaceSingleDollarSignWithDouble(escapedFieldName), StringUtils.EMPTY_STRING));
+
+    while (fieldCounter < recordSchema.getFields().size()) {
+      String chunkMethodName = "customDecodeIfChunk" + chunkCounter;
```

Comment:
nit - why call the method "customDecodeIfChunk0()" ?
as oposed to customDecodeChunk0() ? what does the "if" stand for?

Reply:
![Screenshot 2023-08-16 at 4 35 08 PM](https://github.com/linkedin/avro-util/assets/98059772/9ff5429e-81c5-4320-ad6f-e74cf9457bac)

The generated method has `if` and `else` chunks.